assets/src/js/frontend/give-donations.js

```diff
@@ -604,6 +608,22 @@ Give.form = {
 			);
 		},
 
+		/**
+		 * Check if nonce old or not
+		 *
+		 * @since 2.1.3
+		 * @param {object} $form
+		 *
+		 * @return {boolean}
+		 */
+		isNonceOld: function ($form) {
+			const $nonceField = jQuery('#_wpnonce', $form),
+				nonceTime = parseInt( $nonceField.data('time') ) + parseInt( $nonceField.data('nonce-life') ) - 600,
```

Comment:
@DevinWalker Logic will only reset nonce when nonce reach to its closest possible expire date. I take `30 minute` as threshold time to process donation by donor after loading donation page.